src/mlpack/core/dists/gaussian_distribution.cpp

```diff
@@ -39,8 +39,17 @@ void GaussianDistribution::Covariance(arma::mat&& covariance)
 void GaussianDistribution::FactorCovariance()
 {
   // On Armadillo < 4.500, the "lower" option isn't available.
-  covLower = arma::chol(covariance, "lower");
+  arma::mat sym_cholmatrix = arma::symmatu(covariance);
+  arma::mat covLower;
 
+  // Check if Cholesky Decomposition exists, if not raise an error log.
+  if (sym_cholmatrix.is_empty())
+  {
+    Log::Warn << "Cholesky Decomposition failed as Matrix is Empty ."
+          << std::endl;
+    return;
+  }
+  arma::chol(covLower, sym_cholmatrix, "lower");
```

Comment:
where is the check for chol() failure?  
hint: it returns a bool set to false if it fails.  
documentation: http://arma.sourceforge.net/docs.html#chol